source/developers-guide/rest-api/plugin-api-extension/index.md

```diff
@@ -0,0 +1,557 @@
+---
+layout: default
+title: Create your own REST API endpoint
+github_link: developers-guide/rest-api/plugin-api-extension/index.md
+shopware_version: 5.2.0
+indexed: true
+tags:
+  - extend
+  - api
+  - Plugin
+group: Developer Guides
+subgroup: REST API
+menu_title: Create your own endpoint
+menu_order: 130
+---
+
+<div class="toc-list"></div>
+
+## Introduction
+This article describes how to extend the REST API and create an API endpoint. We create an exmaple plugin which 
+provides functions for managing manufacturer.
+
+Normally every basic API resource contains of two parts:
+* a controller which handles the different request types(POST, GET, PUT, DELETE)
+* the actual resource that takes care of the [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete)
+operations
+
+### The File and directory structure
+<img src="img/file_structure.jpg" alt="API endpoint file structure"/>
+
+## Plugin files
+For our REST API example we only need a few files. For more information about necessary files and the 5.2 plugin system
+see the [5.2 plugin guide](/developers-guide/plugin-system).
+
+### plugin.xml
+contains the base information about the plugin, the label in englisch and german, the plugin version and the required
+shopware version.
+
+```
+<?xml version="1.0" encoding="utf-8"?>
+<plugin xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../engine/Shopware/Components/Plugin/schema/plugin.xsd">
+    <label lang="de">Swag Hersteller API</label>
+    <label lang="en">Swag Producer API</label>
+    <version>1.0.0</version>
+    <compatibility minVersion="5.2.0"/>
+</plugin>
+
+```
+
+### SwagProducerApi.php
+
+```
+<?php
+
+namespace SwagProducerApi;
+
+use Shopware\Components\Plugin;
+
+class SwagProducerApi extends Plugin
+{
+    /**
+     * @inheritdoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Enlight_Controller_Dispatcher_ControllerPath_Api_Producer' => 'onGetProducerApiController',
+            'Enlight_Controller_Front_StartDispatch' => 'onEnlightControllerFrontStartDispatch'
+        ];
+    }
+
+    /**
+     * @return string
+     */
+    public function onGetProducerApiController()
+    {
+        return $this->getPath() . '/Controllers/Api/Producer.php';
+    }
+
+    /**
+     *
+     */
+    public function onEnlightControllerFrontStartDispatch()
+    {
+        $this->container->get('loader')->registerNamespace('Shopware\Components', $this->getPath() . '/Components/');
+    }
+}
+```
+This is our plugin bootstrap which subscribes to two events. For the one thing it uses the
+`Enlight_Controller_Dispatcher_ControllerPath_Api` event to register the API controller and for 
+the other thing it uses the `Enlight_Controller_Front_StartDispatch` event to register our resource to Shopware.
```

Comment:
I would add a reason for registering an additional namespace: we have to that, so the API Manager class could load your recourse with `\Shopware\Components\Api\Manager::getResource('Producer')`